xmppserver/src/main/java/org/jivesoftware/openfire/commands/admin/user/GetUserRoster.java

```diff
@@ -0,0 +1,200 @@
+        for ( final String accountjid : data.get( "accountjids" ))
+        {
+            JID account;
+            try
+            {
+                account = new JID( accountjid );
+
+                if ( !XMPPServer.getInstance().isLocal( account ) )
+                {
+                    note.addAttribute( "type", "error" );
+                    note.setText(LocaleUtils.getLocalizedString("commands.admin.user.getuserroster.note.jid-not-local", List.of(accountjid), preferredLocale));
+                    requestError = true;
+                }
+                else
+                {
+                    rosters.put(account, rosterManager.getRoster(account.getNode()));
+                }
+            }
+            catch ( NullPointerException npe )
+            {
+                note.addAttribute( "type", "error" );
+                note.setText(LocaleUtils.getLocalizedString("commands.admin.user.getuserroster.note.jid-required", preferredLocale));
+                requestError = true;
+            }
+            catch (IllegalArgumentException npe)
+            {
+                note.addAttribute( "type", "error" );
+                note.setText(LocaleUtils.getLocalizedString("commands.admin.user.getuserroster.note.jid-invalid", preferredLocale));
+                requestError = true;
+            }
+            catch ( UserNotFoundException e )
+            {
+                note.addAttribute( "type", "error" );
+                note.setText(LocaleUtils.getLocalizedString("commands.admin.user.getuserroster.note.user-does-not-exist", List.of(accountjid), preferredLocale));
+                requestError = true;
+            }
+        }
+
+        if (rosters.size() > 1) {
+            note.addAttribute( "type", "error" );
+            note.setText(LocaleUtils.getLocalizedString("commands.admin.user.getuserroster.note.cannot-return-multiple", preferredLocale));
+            requestError = true;
+        }
+
+        if ( requestError )
+        {
+            // We've collected all errors. Return without applying changes.
+            return;
+        }
+
+        // No errors.
+        for (final Map.Entry<JID, Roster> entry : rosters.entrySet()) {
+            final JID owner = entry.getKey();
+            final Roster roster = entry.getValue();
+
+            DataForm form = new DataForm(DataForm.Type.result);
+
+            FormField field = form.addField();
+            field.setType(FormField.Type.hidden);
+            field.setVariable("FORM_TYPE");
+            field.addValue("http://jabber.org/protocol/admin");
+
+            field = form.addField();
+            field.setType(FormField.Type.jid_multi);
+            field.setVariable("accountjids");
+            field.addValue(owner);
+
+            final Element rosterElement = roster.getReset().getElement()
+                .element(QName.get("query", "jabber:iq:roster"));
+
+            if (rosterElement != null) {
+                form.getElement().add(rosterElement);
```

Comment:
This line raises an error when called

```
2024.01.14 16:00:16.160 ERROR [nioEventLoopGroup-3-15]: org.jivesoftware.openfire.handler.IQHandler - Internal server error
org.dom4j.IllegalAddException: The node "org.dom4j.tree.DefaultElement@49f4ad8 [Element: <query uri: jabber:iq:roster attributes: []/>]" could not be added to the element "x" because: The Node already has an existing parent of "iq"
    at org.dom4j.tree.AbstractElement.addNode(AbstractElement.java:1376) ~[dom4j-2.1.4.jar:?]
    at org.dom4j.tree.AbstractElement.add(AbstractElement.java:910) ~[dom4j-2.1.4.jar:?]
    at org.jivesoftware.openfire.commands.admin.user.GetUserRoster.execute(GetUserRoster.java:151) ~[xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
```

Reply:
Ah, that element needs to be detached from its previous parent. I've pushed a change that should take care of this.